### PR TITLE
Replace Travis CI by GitHub workflow

### PR DESCRIPTION
**Pull request by sbesson**
- use standard build job to build, run htmlproofer and upload artifact
- add dependent release job to consume the artifact and upload as release asset

### CODE REVIEW

**Comment by sbesson** [comment]
See https://github.com/sbesson/idr.openmicroscopy.org/releases/tag/2020.11.23 and https://github.com/sbesson/idr.openmicroscopy.org/actions/runs/378862175 for an application of this workflow on a tag.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+name: Build
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * 0'
```
Comment (manics) [inline .github/workflows/build.yml:8]: Is this needed?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build static website
+        run: docker run --rm -v $PWD:/srv/jekyll -eJEKYLL_UID=$UID jekyll/builder:pages jekyll build --config _config.yml,_prod.yml
+      - name: Run htmlproofer
+        run: docker run --rm -v $PWD/_site:/site/about jekyll/builder:pages /usr/gem/bin/htmlproofer /site --url-ignore "/jupyter,/webclient/,/cell/,/tissue/,/,"
```
Comment (manics) [inline .github/workflows/build.yml:18]: This replaces https://github.com/IDR/idr.openmicroscopy.org/blob/ef5ab7ff7ce693ccb2c73dacfd233a48b0fe19ee/build.sh

Advantage of keeping the script is it's easy to run locally. Disadvantage is the commands are duplicated in two places.

Comment (sbesson) [reply .github/workflows/build.yml:18]: Thanks and agreeed. I'll work on a second version that consumes the script directly although I'll need to drop some of the arguments (`-it`) which are simply not compatible with the GH environment.

File: .github/workflows/build.yml (hunk 1 above)
Comment (sbesson) [reply .github/workflows/build.yml:8]: The goal was to be able to detect asynchronously whether 3d-party links are broken. Maybe we update this repository often enough that we can omit the cron configuration for now.

**Comment by sbesson** [comment]
The last commit update the workflow to be more agnostic to the static website, delegating the build to a `build.sh` script and using `github.event.repository.name` to retrieve the name of the repository. Note a downside of this approach is that it does not apply to cron executions.

See https://github.com/sbesson/idr.openmicroscopy.org/actions/runs/381305125, https://github.com/sbesson/idr.openmicroscopy.org/actions/runs/381311418 and https://github.com/sbesson/idr.openmicroscopy.org/releases/tag/2020.11.24 for the result of the last commit with and without a tag.

**Review by manics — APPROVED** [review]
Thanks, this LGTM

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -4,25 +4,23 @@ name: Build
...
       - name: Build static website
-        run: docker run --rm -v $PWD:/srv/jekyll -eJEKYLL_UID=$UID jekyll/builder:pages jekyll build --config _config.yml,_prod.yml
-      - name: Run htmlproofer
-        run: docker run --rm -v $PWD/_site:/site/about jekyll/builder:pages /usr/gem/bin/htmlproofer /site --url-ignore "/jupyter,/webclient/,/cell/,/tissue/,/,"
+        run: ./build.sh
+        env:
+          UID: 1001
       - name: Create website archive
-        run: tar -zcvf idr.openmicroscopy.org.tar.gz -C _site ./
+        run: tar -zcvf ${{ github.event.repository.name }}.tar.gz -C _site ./
```
Comment (joshmoore) [inline .github/workflows/build.yml:18]: This is so that the workflow can be a template?

Comment (sbesson) [reply .github/workflows/build.yml:18]: Yes that's the idea. https://github.com/ome/www.openmicroscopy.org is the next one where I'd like to test this

Comment (joshmoore) [reply .github/workflows/build.yml:18]: Cool. Then I guess if you wanted to keep the cronjob you could introduce a variable.

Comment (sbesson) [reply .github/workflows/build.yml:18]: Agreed. The alternative is to use consume `github.repository` (https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions) and do some string manipulation to extract the name

**Pull request merged**